src/features/singleRight/delegate/ChooseRightsPage/ChooseRightsPage.tsx

```diff
@@ -1,17 +1,239 @@
 import { useTranslation } from 'react-i18next';
 import { PersonIcon } from '@navikt/aksel-icons';
+import { Alert, Button, Chip, Ingress, Paragraph, Popover } from '@digdir/design-system-react';
+import { useNavigate } from 'react-router-dom';
+import { useMemo, useState } from 'react';
 
-import { Page, PageContainer, PageContent, PageHeader } from '@/components';
+import { DualElementsContainer, Page, PageContainer, PageContent, PageHeader } from '@/components';
+import { useAppDispatch, useAppSelector } from '@/rtk/app/hooks';
+import { SingleRightPath } from '@/routes/paths';
+import { useMediaQuery } from '@/resources/hooks';
+import {
+  type ChosenService,
+  removeServiceResource,
+} from '@/rtk/features/singleRights/singleRightsSlice';
+import { getSingleRightsErrorCodeTextKey } from '@/resources/utils/errorCodeUtils';
+
+import { ResourceActionBar } from '../ChooseServicePage/ResourceActionBar/ResourceActionBar';
+
+import classes from './ChooseRightsPage.module.css';
+
+interface DelegationResourceDTO {
+  title: string | undefined;
+  serviceIdentifier: string | undefined;
+  rightKey: string;
+  checked: boolean;
+}
 
 export const ChooseRightsPage = () => {
   const { t } = useTranslation();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const [checkedStates, setCheckedStates] = useState<Record<string, boolean>>({});
+  const isSm = useMediaQuery('(max-width: 768px)');
+  const rightsToBeDelegated: DelegationResourceDTO[] = [];
+
+  const chosenServices = useAppSelector((state) => state.singleRightsSlice.chosenServices);
+
+  const delegableChosenServices = chosenServices.filter((s) => s.status !== 'NotDelegable');
+
+  const onRemove = (identifier: string | undefined) => {
+    void dispatch(removeServiceResource(identifier));
+  };
+
+  const onConfirm = () => {
+    setPopoverOpen(!popoverOpen);
+    const filteredList = rightsToBeDelegated.filter((right) => right.checked);
+    const listWithoutChecked = filteredList.map(({ checked, ...rest }) => rest);
+  };
+
+  const handleToggleChecked = (serviceIdentifier: string, rightKey: string) => {
+    setCheckedStates((prevCheckedStates) => ({
+      ...prevCheckedStates,
+      [`${serviceIdentifier}-${rightKey}`]: !prevCheckedStates[`${serviceIdentifier}-${rightKey}`],
+    }));
+  };
+
+  const sortedServiceResources = useMemo(() => {
+    return [...delegableChosenServices]?.sort((a, b) => {
+      const isPartiallyDelegableA = a.status === 'PartiallyDelegable';
+      const isPartiallyDelegableB = b.status === 'PartiallyDelegable';
+
+      if (isPartiallyDelegableA && !isPartiallyDelegableB) {
+        return -1;
+      }
+      if (!isPartiallyDelegableA && isPartiallyDelegableB) {
+        return 1;
+      }
+
+      return a.service?.title.localeCompare(b.service.title);
+    });
+  }, [delegableChosenServices]);
+
+  const serviceResourcesActionBars = useMemo(() => {
```

Comment:
This method is so long, and so complex, it's frustratingly hard to read.
Maybe you can split it up, or even separate it into a separate local component to make it more manageable?